tools/getrealdeps.rb

```diff
@@ -1,121 +1,141 @@
 #!/usr/bin/env ruby
-# getrealdeps for Chromebrew version 1.2
+# getrealdeps for Chromebrew version 1.3
 # Author: Satadru Pramanik (satmandu) satadru at gmail dot com
 require 'fileutils'
 
-# Exit quickly if an invalid package name is given.
-if ARGV[0].nil? || ARGV[0].empty? || ARGV[0].include?('#')
-  puts 'Getrealdeps checks for the dependencies of a package.'
-  puts 'Dependencies are added if the package file is missing them.'
-  puts 'Usage: getrealdeps.rb <packagename>'
-  exit 1
-end
-
-pkg = ARGV[0].chomp('.rb')
-
-CREW_PREFIX = ENV['CREW_PREFIX'] || `crew const CREW_PREFIX`.split('=')[1].chomp
-CREW_LIB_PREFIX = ENV['CREW_LIB_PREFIX'] || `crew const CREW_LIB_PREFIX`.split('=')[1].chomp
-
-# Package needs to be installed for package filelist to be populated.
-unless File.exist? "#{CREW_PREFIX}/etc/crew/meta/#{pkg}.filelist"
-  puts "Installing #{pkg} because it is not installed."
-  system("yes | crew install #{pkg}")
-end
-
-unless File.exist?("#{CREW_PREFIX}/etc/crew/meta/#{pkg}.filelist")
-  puts "Package #{pkg} either does not exist or does not contain any libraries."
-  exit 1
-end
-
-# Speed up grep.
-ENV['LC_ALL'] = 'C'
-
-# Install grep if a functional local copy does not exist.
-if system('grep --version > /dev/null 2>&1')
-  GREP = 'grep'
+@crew_local_repo_root = `git rev-parse --show-toplevel 2> /dev/null`.chomp
+# When invoked from crew, pwd is CREW_DEST_DIR, so @crew_local_repo_root
+# is empty.
+if @crew_local_repo_root.to_s.empty?
+  require_relative '../lib/const'
 else
-  system('crew install grep')
-  GREP = "#{CREW_PREFIX}/bin/grep"
+  require File.join(@crew_local_repo_root, 'lib/const')
```

Comment:
`/lib/const`?

Reply:
File.join handles this just fine adding necessary slashes.

Reply:
https://ruby-doc.org/3.2.2/File.html#method-c-join

Reply:
These functions try to be too smart.  The more you know.

Reply:
It's great because it adds a slash without adding an extra slash if one of the strings ends in a slash!